### ISSUE TITLE
Improve error message for wrong port provided

### ISSUE THREAD

**Issue by cdrage** [openshift/odo#1876]
[kind/Enhancement]
<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the chat and talk to us if you have a question rather than a bug or feature request.
  
The chat room is at: https://chat.openshift.io/developers/channels/odo

Thanks for understanding, and for contributing to the project!

-->

## Which functionality do you think we should update/improve?

```sh
~/nodejs-ex  master ✗                                                                                                                                                                                                                         255d ◒  ⍉
▶ odo url create foobar --port 9999                           
 ✗  given port 9999 is not exposed on given component      
```

IMO we should update the output to show WHAT ports are available to the user (what can actually be exposed...)

## Why is this needed? 

Better UX / user experience.

**Comment by surajnarwade** [comment]
this will be resolved in https://github.com/openshift/odo/pull/1828

**Comment by surajnarwade** [comment]
Closing issue since this is fixed in  #1828  & it is merged :tada:
